### ISSUE TITLE
Nested queries with duplicate column names fail

### ISSUE THREAD

**Issue by tlrobinson** [metabase/metabase#10511]
e.x. these breakouts are both `CREATED_AT`, which works fine:

```
    { 'source-table': 2,
      aggregation: [ [ 'count' ] ],
      breakout:
       [ [ 'datetime-field', [ 'field-id', 15 ], 'month' ],
         [ 'datetime-field', [ 'fk->', [ 'field-id', 11 ], [ 'field-id', 7 ] ], 'month' ] ] },
```

but if you nested the query, e.x. with a post-aggregation filter, it returns `Duplicate column name "CREATED_AT"`

```
   { 'source-query':
      { 'source-table': 2,
        aggregation: [ [ 'count' ] ],
        breakout:
         [ [ 'datetime-field', [ 'field-id', 15 ], 'month' ],
           [ 'datetime-field', [ 'fk->', [ 'field-id', 11 ], [ 'field-id', 7 ] ], 'month' ] ] },
     filter:
      [ '>', [ 'field-literal', 'count', 'type/Integer' ], '3456' ] },
```

Full logs:

```
[backend] 08-06 17:28:07 DEBUG middleware.log :: POST /api/dataset 200 [ASYNC: completed] 135 ms (17 DB calls) Jetty threads: 3/50 (3 idle, 0 queued) (106 total active threads) Queries in flight: 0
[backend] JdbcSQLException:
[backend]  Message: Duplicate column name "CREATED_AT"; SQL statement:
[backend] -- Metabase:: userID: 1 queryType: MBQL queryHash: b6121913705d8d057cc482e8643d916d73aa578a7944b4a4dc6adeb1677217d3
[backend] SELECT "source"."CREATED_AT" AS "CREATED_AT", "source"."CREATED_AT_2" AS "CREATED_AT_2", "source"."count" AS "count" FROM (SELECT parsedatetime(formatdatetime("PUBLIC"."ORDERS"."CREATED_AT", 'yyyyMM'), 'yyyyMM') AS "CREATED_AT", parsedatetime(formatdatetime("PRODUCTS__via__PRODUCT_ID"."CREATED_AT", 'yyyyMM'), 'yyyyMM') AS "CREATED_AT", count(*) AS "count" FROM "PUBLIC"."ORDERS" LEFT JOIN "PUBLIC"."PRODUCTS" "PRODUCTS__via__PRODUCT_ID" ON "PUBLIC"."ORDERS"."PRODUCT_ID" = "PRODUCTS__via__PRODUCT_ID"."ID" GROUP BY parsedatetime(formatdatetime("PUBLIC"."ORDERS"."CREATED_AT", 'yyyyMM'), 'yyyyMM'), parsedatetime(formatdatetime("PRODUCTS__via__PRODUCT_ID"."CREATED_AT", 'yyyyMM'), 'yyyyMM') ORDER BY parsedatetime(formatdatetime("PUBLIC"."ORDERS"."CREATED_AT", 'yyyyMM'), 'yyyyMM') ASC, parsedatetime(formatdatetime("PRODUCTS__via__PRODUCT_ID"."CREATED_AT", 'yyyyMM'), 'yyyyMM') ASC) "source" WHERE "source"."count" > ? LIMIT 2000 [42121-197]
[backend]  SQLState: 42S21
[backend]  Error Code: 42121
[backend] 08-06 17:30:39 ERROR sql-jdbc.execute :: nil
[backend] 08-06 17:30:39 WARN middleware.process-userland-query :: Query failure {:status :failed,
[backend]  :class java.lang.Exception,
[backend]  :error "Duplicate column name \"CREATED_AT\"",
[backend]  :stacktrace
[backend]  ("--> driver.sql_jdbc.execute$do_with_try_catch.invokeStatic(execute.clj:221)"
[backend]   "driver.sql_jdbc.execute$do_with_try_catch.invoke(execute.clj:211)"
[backend]   "driver.sql_jdbc.execute$execute_query.invokeStatic(execute.clj:285)"
[backend]   "driver.sql_jdbc.execute$execute_query.invoke(execute.clj:279)"
[backend]   "driver.sql_jdbc$eval75638$fn__75639.invoke(sql_jdbc.clj:44)"
[backend]   "query_processor$eval50532$execute_query__50537$fn__50538.invoke(query_processor.clj:70)"
[backend]   "query_processor$eval50532$execute_query__50537.invoke(query_processor.clj:64)"
[backend]   "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__39481.invoke(mbql_to_native.clj:38)"
[backend]   "query_processor.middleware.annotate$result_rows_maps__GT_vectors$fn__41772.invoke(annotate.clj:537)"
[backend]   "query_processor.middleware.annotate$add_column_info$fn__41678.invoke(annotate.clj:481)"
[backend]   "query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__42779.invoke(cumulative_aggregations.clj:57)"
[backend]   "query_processor.middleware.resolve_joins$resolve_joins$fn__47232.invoke(resolve_joins.clj:184)"
[backend]   "query_processor.middleware.limit$limit$fn__43836.invoke(limit.clj:19)"
[backend]   "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__50347.invoke(results_metadata.clj:86)"
[backend]   "query_processor.middleware.format_rows$format_rows$fn__43816.invoke(format_rows.clj:26)"
[backend]   "query_processor.middleware.add_dimension_projections$add_remapping$fn__40195.invoke(add_dimension_projections.clj:234)"
[backend]   "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__40849.invoke(add_source_metadata.clj:101)"
[backend]   "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__47284.invoke(resolve_source_table.clj:48)"
[backend]   "query_processor.middleware.add_row_count_and_status$add_row_count_and_status$fn__40698.invoke(add_row_count_and_status.clj:16)"
[backend]   "query_processor.middleware.driver_specific$process_query_in_context$fn__43012.invoke(driver_specific.clj:12)"
[backend]   "query_processor.middleware.resolve_driver$resolve_driver$fn__46880.invoke(resolve_driver.clj:23)"
[backend]   "query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__42123$fn__42124.invoke(bind_effective_timezone.clj:9)"
[backend]   "util.date$call_with_effective_timezone.invokeStatic(date.clj:88)"
[backend]   "util.date$call_with_effective_timezone.invoke(date.clj:77)"
[backend]   "query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__42123.invoke(bind_effective_timezone.clj:8)"
[backend]   "query_processor.middleware.store$initialize_store$fn__50386$fn__50387.invoke(store.clj:11)"
[backend]   "query_processor.store$do_with_store.invokeStatic(store.clj:46)"
[backend]   "query_processor.store$do_with_store.invoke(store.clj:40)"
[backend]   "query_processor.middleware.store$initialize_store$fn__50386.invoke(store.clj:10)"
[backend]   "query_processor.middleware.async$async__GT_sync$fn__39376.invoke(async.clj:23)"
[backend]   "query_processor.middleware.async_wait$runnable$fn__41837.invoke(async_wait.clj:89)"),
[backend]  :query
[backend]  {:query
[backend]   {:source-query {:source-table 2, :aggregation [["count"]], :breakout [["datetime-field" ["field-id" 15] "month"] ["datetime-field" ["fk->" ["field-id" 11] ["field-id" 7]] "month"]]},
[backend]    :filter [">" ["field-literal" "count" "type/Integer"] "3456"]},
[backend]   :type "query",
[backend]   :parameters [],
[backend]   :async? true,
[backend]   :middleware {:add-default-userland-constraints? true, :userland-query? true},
[backend]   :info
[backend]   {:executed-by 1,
[backend]    :context :ad-hoc,
[backend]    :card-id nil,
[backend]    :nested? false,
[backend]    :query-hash [-74, 18, 25, 19, 112, 93, -115, 5, 124, -60, -126, -24, 100, 61, -111, 109, 115, -86, 87, -118, 121, 68, -76, -92, -36, 106, -34, -79, 103, 114, 23, -45]},
[backend]   :constraints {:max-results 10000, :max-results-bare-rows 2000}},
[backend]  :preprocessed
[backend]  {:query
[backend]   {:source-query
[backend]    {:source-table 2,
[backend]     :aggregation [[:aggregation-options [:count] {:name "count"}]],
    :breakout [[:datetime-field [:field-id 15] :month] [:datetime-field [:joined-field "PRODUCTS__via__PRODUCT_ID" [:field-id 7]] :month]],
[backend]     :order-by [[:asc [:datetime-field [:field-id 15] :month]] [:asc [:datetime-field [:joined-field "PRODUCTS__via__PRODUCT_ID" [:field-id 7]] :month]]],
[backend]     :joins [{:strategy :left-join, :source-table 1, :alias "PRODUCTS__via__PRODUCT_ID", :fk-field-id 11, :condition [:= [:field-id 11] [:joined-field "PRODUCTS__via__PRODUCT_ID" [:field-id 4]]]}]},
[backend]    :filter [:> [:field-literal "count" :type/Integer] [:value "3456" nil]],
[backend]    :source-metadata
[backend]    [{:table_id 2,
[backend]      :special_type :type/CreationTimestamp,
[backend]      :unit :month,
[backend]      :name "CREATED_AT",
[backend]      :settings nil,
[backend]      :id 15,
[backend]      :display_name "Created At",
[backend]      :fingerprint {:global {:distinct-count 10001, :nil% 0.0}, :type {:type/DateTime {:earliest "2016-05-01T01:56:13.352Z", :latest "2020-04-19T21:07:15.657Z"}}},
[backend]      :base_type :type/DateTime}
[backend]     {:table_id 1,
[backend]      :special_type :type/CreationTimestamp,
[backend]      :unit :month,
[backend]      :name "CREATED_AT_2",
[backend]      :settings nil,
[backend]      :id 7,
[backend]      :display_name "Products → Created At",
[backend]      :fingerprint {:global {:distinct-count 200, :nil% 0.0}, :type {:type/DateTime {:earliest "2016-04-27T02:29:55.147Z", :latest "2019-04-15T20:34:19.931Z"}}},
[backend]      :base_type :type/DateTime}
[backend]     {:name "count", :display_name "Count", :base_type :type/Integer, :special_type :type/Number}],
[backend]    :fields [[:field-literal "CREATED_AT" :type/DateTime] [:field-literal "CREATED_AT_2" :type/DateTime] [:field-literal "count" :type/Integer]],
[backend]    :limit 2000},
[backend]   :type :query,
[backend]   :database 1,
[backend]   :middleware {:add-default-userland-constraints? true, :userland-query? true},
[backend]   :info
[backend]   {:executed-by 1,
[backend]    :context :ad-hoc,
[backend]    :nested? false,
[backend]    :query-hash [-74, 18, 25, 19, 112, 93, -115, 5, 124, -60, -126, -24, 100, 61, -111, 109, 115, -86, 87, -118, 121, 68, -76, -92, -36, 106, -34, -79, 103, 114, 23, -45]},
[backend]   :constraints {:max-results 10000, :max-results-bare-rows 2000},
[backend]   :driver :h2},
[backend]  :native
[backend]  {:query
[backend]   "SELECT \"source\".\"CREATED_AT\" AS \"CREATED_AT\", \"source\".\"CREATED_AT_2\" AS \"CREATED_AT_2\", \"source\".\"count\" AS \"count\" FROM (SELECT parsedatetime(formatdatetime(\"PUBLIC\".\"ORDERS\".\"CREATED_AT\", 'yyyyMM'), 'yyyyMM') AS \"CREATED_AT\", parsedatetime(formatdatetime(\"PRODUCTS__via__PRODUCT_ID\".\"CREATED_AT\", 'yyyyMM'), 'yyyyMM') AS \"CREATED_AT\", count(*) AS \"count\" FROM \"PUBLIC\".\"ORDERS\" LEFT JOIN \"PUBLIC\".\"PRODUCTS\" \"PRODUCTS__via__PRODUCT_ID\" ON \"PUBLIC\".\"ORDERS\".\"PRODUCT_ID\" = \"PRODUCTS__via__PRODUCT_ID\".\"ID\" GROUP BY parsedatetime(formatdatetime(\"PUBLIC\".\"ORDERS\".\"CREATED_AT\", 'yyyyMM'), 'yyyyMM'), parsedatetime(formatdatetime(\"PRODUCTS__via__PRODUCT_ID\".\"CREATED_AT\", 'yyyyMM'), 'yyyyMM') ORDER BY parsedatetime(formatdatetime(\"PUBLIC\".\"ORDERS\".\"CREATED_AT\", 'yyyyMM'), 'yyyyMM') ASC, parsedatetime(formatdatetime(\"PRODUCTS__via__PRODUCT_ID\".\"CREATED_AT\", 'yyyyMM'), 'yyyyMM') ASC) \"source\" WHERE \"source\".\"count\" > ? LIMIT 2000",
[backend]   :params ("3456")},
[backend]  :cause
[backend]  {:class org.h2.jdbc.JdbcSQLException,
[backend]   :error
[backend]   "Duplicate column name \"CREATED_AT\"; SQL statement:\n-- Metabase:: userID: 1 queryType: MBQL queryHash: b6121913705d8d057cc482e8643d916d73aa578a7944b4a4dc6adeb1677217d3\nSELECT \"source\".\"CREATED_AT\" AS \"CREATED_AT\", \"source\".\"CREATED_AT_2\" AS \"CREATED_AT_2\", \"source\".\"count\" AS \"count\" FROM (SELECT parsedatetime(formatdatetime(\"PUBLIC\".\"ORDERS\".\"CREATED_AT\", 'yyyyMM'), 'yyyyMM') AS \"CREATED_AT\", parsedatetime(formatdatetime(\"PRODUCTS__via__PRODUCT_ID\".\"CREATED_AT\", 'yyyyMM'), 'yyyyMM') AS \"CREATED_AT\", count(*) AS \"count\" FROM \"PUBLIC\".\"ORDERS\" LEFT JOIN \"PUBLIC\".\"PRODUCTS\" \"PRODUCTS__via__PRODUCT_ID\" ON \"PUBLIC\".\"ORDERS\".\"PRODUCT_ID\" = \"PRODUCTS__via__PRODUCT_ID\".\"ID\" GROUP BY parsedatetime(formatdatetime(\"PUBLIC\".\"ORDERS\".\"CREATED_AT\", 'yyyyMM'), 'yyyyMM'), parsedatetime(formatdatetime(\"PRODUCTS__via__PRODUCT_ID\".\"CREATED_AT\", 'yyyyMM'), 'yyyyMM') ORDER BY parsedatetime(formatdatetime(\"PUBLIC\".\"ORDERS\".\"CREATED_AT\", 'yyyyMM'), 'yyyyMM') ASC, parsedatetime(formatdatetime(\"PRODUCTS__via__PRODUCT_ID\".\"CREATED_AT\", 'yyyyMM'), 'yyyyMM') ASC) \"source\" WHERE \"source\".\"count\" > ? LIMIT 2000 [42121-197]"}}
[backend]
```

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by flamber** [comment]
Just adding steps, since I had forgotten. On `master` commit `81edcd4`:
1. Custom question > Sample Dataset > Orders
2. Summarize "Count" by "CreatedAt:Month" and "Product->CreatedAt:Month"
3. Filter "Count" greater than `5`

![image](https://user-images.githubusercontent.com/1447303/83358423-b4511280-a373-11ea-9613-03be8d1e99bd.png)

**Comment by sbelak** [comment]
Related to #12649

**Comment by camsaul** [comment]
Fixed by #13648
